modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -476,4 +554,61 @@ private boolean isMultiPartUpload(String request) {
                 || Regex.simpleMatch("PUT /*/*?*uploadId=*", request);
         }
     }
+
```

Comment:
I think this is a nice idea on testing this. 
cc @stu-elastic perhaps this might help us, we could borrow some parts of this